engine/jobmaster/dm/dm_jobmaster.go

```diff
@@ -117,6 +118,7 @@ func (jm *JobMaster) initComponents() error {
 	jm.taskManager = NewTaskManager(taskStatus, jm.metadata.JobStore(), jm.messageAgent, jm.Logger(), jm.MetricFactory())
 	jm.workerManager = NewWorkerManager(jm.ID(), workerStatus, jm.metadata.JobStore(), jm.metadata.UnitStateStore(),
 		jm, jm.messageAgent, jm.checkpointAgent, jm.Logger(), jm.IsS3StorageEnabled())
+	jm.ddlCoordinator = NewDDLCoordinator(jm.ID(), jm.MetaKVClient(), jm.checkpointAgent, jm.metadata.JobStore(), jm.Logger())
```

Comment:
Can we not enable this member if job is using sharding mode? To reduce the chance we meet error.

Reply:
still enable it, but bypassing in the internal 
00f0094